src/SprykerEco/Zed/Payone/Business/Payment/MethodSender/PayonePartialRefundMethodSender.php

```diff
@@ -0,0 +1,257 @@
+            }
+
+            $arrayIt[$key] = PayoneApiConstants::INVOICING_ITEM_TYPE_GOODS;
+            $arrayId[$key] = $itemTransfer->getSku();
+            $arrayPr[$key] = $itemTransfer->getRefundableAmount();
+            $arrayNo[$key] = $itemTransfer->getQuantity();
+            $arrayDe[$key] = $itemTransfer->getName();
+            $arrayVa[$key] = (int)$itemTransfer->getTaxRate();
+            $key++;
+        }
+
+        $container->setIt($arrayIt);
+        $container->setId($arrayId);
+        $container->setPr($arrayPr);
+        $container->setNo($arrayNo);
+        $container->setDe($arrayDe);
+        $container->setVa($arrayVa);
+
+        return $container;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PayonePartialOperationRequestTransfer $payonePartialOperationRequestTransfer
+     * @param \SprykerEco\Zed\Payone\Business\Api\Request\Container\AbstractRequestContainer $container
+     *
+     * @return \SprykerEco\Zed\Payone\Business\Api\Request\Container\AbstractRequestContainer
+     */
+    protected function preparePartialRefundExpenses(
+        PayonePartialOperationRequestTransfer $payonePartialOperationRequestTransfer,
+        AbstractRequestContainer $container
+    ): AbstractRequestContainer {
+        $arrayIt = $container->getIt() ?? [];
+        $arrayId = $container->getId() ?? [];
+        $arrayPr = $container->getPr() ?? [];
+        $arrayNo = $container->getNo() ?? [];
+        $arrayDe = $container->getDe() ?? [];
+        $arrayVa = $container->getVa() ?? [];
+
+        $key = count($arrayId) + 1;
+
+        foreach ($payonePartialOperationRequestTransfer->getRefund()->getExpenses() as $expenseTransfer) {
+            $expenseType = PayoneApiConstants::INVOICING_ITEM_TYPE_HANDLING;
+            if ($expenseTransfer->getType() === ShipmentConfig::SHIPMENT_EXPENSE_TYPE) {
+                $expenseType = PayoneApiConstants::INVOICING_ITEM_TYPE_SHIPMENT;
+            }
+
+            $arrayIt[$key] = $expenseType;
+            $arrayId[$key] = $expenseType;
+            $arrayPr[$key] = $expenseTransfer->getRefundableAmount();
+            $arrayNo[$key] = $expenseTransfer->getQuantity();
+            $arrayDe[$key] = $expenseTransfer->getName();
+            $arrayVa[$key] = (int)$expenseTransfer->getTaxRate();
+            $key++;
+        }
+
+        $container->setIt($arrayIt);
+        $container->setId($arrayId);
+        $container->setPr($arrayPr);
+        $container->setNo($arrayNo);
+        $container->setDe($arrayDe);
+        $container->setVa($arrayVa);
+
+        return $container;
+    }
+
+    /**
+     * @param \Orm\Zed\Payone\Persistence\SpyPaymentPayoneApiLog $apiLogEntity
+     * @param \SprykerEco\Zed\Payone\Business\Api\Response\Container\RefundResponseContainer $responseContainer
+     *
+     * @return void
+     */
+    protected function updateApiLogAfterRefund(SpyPaymentPayoneApiLog $apiLogEntity, RefundResponseContainer $responseContainer)
```

Comment:
* Extract to a `AbstractPayoneMethodSender`.
* Used in the `PayoneRefundMethodSender` also.

Reply:
It's required only for 2 classes but in AbstractPayoneMethodSender it will be available in 8. Also all `updateApiLog` methods from other 6 classes should be moved here and they will be available in classes where they should not